### PR TITLE
First PR

### PR DESCRIPTION
**Pull request by iffifan**

### CODE REVIEW

File: config/winnie-client.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+return [
```
Comment (kevzz1994) [inline config/winnie-client.php:3]: Remove the default values, endpoints should not be publicly visible

File: src/Http/Middleware/CheckWinnieUser.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        $user = $request->user()->winnie ?? WinnieClient::withToken($request->user()->access_token)->getUser();
+
+        if (is_array($user) && array_key_exists('roles', $user)) {
+            $user = $request->user()->setWinnieUser($user);
+        } elseif ($user instanceof Authenticatable) {
+            $user = $user->winnie;
+        }
+
+        if ($user instanceof User) {
+            if ( ! $user->hasRole($role)) {
```
Comment (kevzz1994) [inline src/Http/Middleware/CheckWinnieUser.php]: Fix spacing

File: tests/Dummy/User.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+namespace Iffifan\WinnieClient\Tests\Dummy;
+
+use Iffifan\WinnieClient\Models\Traits\HasWinnieUser;
+use Iffifan\WinnieClient\Models\User as WinnieUser;
+
+class User extends \Illuminate\Foundation\Auth\User
+{
+    use HasWinnieUser;
+
+    public static function updateOrCreate(array $criteria, array $data): User
```
Comment (kevzz1994) [inline tests/Dummy/User.php:12]: This doesn't do anything?

**Review by kevzz1994 — CHANGES_REQUESTED** [review]

File: tests/Dummy/User.php (hunk 3 above)
Comment (iffifan) [reply tests/Dummy/User.php:12]: I had to mimic updateOrCreate method of model class of laravel.

File: src/Http/Middleware/CheckWinnieUser.php (hunk 2 above)
Comment (iffifan) [reply src/Http/Middleware/CheckWinnieUser.php]: Updated

File: config/winnie-client.php (hunk 1 above)
Comment (iffifan) [reply config/winnie-client.php:3]: Updated

**Review by kevzz1994 — APPROVED** [review]

**Pull request merged**